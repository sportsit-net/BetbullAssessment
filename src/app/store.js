import { configureStore } from '@reduxjs/toolkit';
import loginReducer from '../components/Login/loginSlice';
import gameReducer from '../components/Game/gameSlice';
import scoreCardReducer from '../components/ScoreCard/scorecardSlice';

export default configureStore({
  reducer: {
    login: loginReducer,
    game: gameReducer,
    scorecard:scoreCardReducer
  },
});
