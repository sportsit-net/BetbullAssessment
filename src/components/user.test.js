import React from 'react'
import UserForm from './userForm';
import { shallow,configure,mount} from 'enzyme';
import {Provider} from 'react-redux';
 import store from './../store'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';


configure({ adapter: new Adapter() });
describe('<UserForm /> with no props',()=>{
    const wrapper = mount(<Provider store={store}><UserForm /></Provider>);
     const loginForm=wrapper.find("[data-id='loginForm']");
     it('checks input validation',()=>{
const userValue=loginForm.find('input');
userValue.instance().value="sample_User"
expect(userValue.instance().value.length).toBe(11);
     });
     
})