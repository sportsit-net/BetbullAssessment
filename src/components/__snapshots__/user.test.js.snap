// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserForm /> with no props should match the snapshot 1`] = `"<form class=\\"userForm SM-formControl-group mb-4  hello\\"><div class=\\"SM-formControl form-row\\"><label class=\\"SM-formLabel SM-txt-white form-label\\">Your Name</label><input type=\\"text\\" data-id=\\"userName\\" placeholder=\\"Enter Name\\" class=\\"SM-input form-control\\"/></div><button type=\\"submit\\" class=\\"btn SM-btnPrimary SM-btnMedium w-100 mt-4 b-0 playButton \\">Lets Play</button></form>"`;

exports[`User Component Should render without errors 1`] = `"<form class=\\"userForm SM-formControl-group mb-4  hello\\"><div class=\\"SM-formControl form-row\\"><label class=\\"SM-formLabel SM-txt-white form-label\\">Your Name</label><input type=\\"text\\" data-id=\\"userName\\" placeholder=\\"Enter Name\\" class=\\"SM-input form-control\\"/></div><button type=\\"submit\\" class=\\"btn SM-btnPrimary SM-btnMedium w-100 mt-4 b-0 playButton \\">Lets Play</button></form>"`;
